crates/nostr/src/nips/nip21.rs

```diff
@@ -179,10 +184,67 @@ impl Nip21 {
     }
 }
 
+/// Extract `nostr:` URIs from a text
```

Comment:
Since I want to use this to add tags to my events, it's helpful to know that I should deduplicate the result vector myself

```suggestion
/// Extract `nostr:` URIs from a text
///
/// The returned vector maintains the same order as their occurrences in the `input`. 
/// The result is not deduplicated, meaning that if a URI appears multiple times in the input, 
/// it will appear the same number of times in the output.
```